diff -up grpc-1.81.1/setup.py.2~ grpc-1.81.1/setup.py
--- grpc-1.81.1/setup.py.2~	2026-06-04 23:58:26.000000000 +0200
+++ grpc-1.81.1/setup.py	2026-06-09 06:49:41.924226296 +0200
@@ -272,7 +272,7 @@ if EXTRA_ENV_COMPILE_ARGS is None:
         EXTRA_ENV_COMPILE_ARGS += " /Zc:preprocessor"
     elif "linux" in sys.platform:
         # GCC by defaults uses C17 so only C++17 needs to be specified.
-        EXTRA_ENV_COMPILE_ARGS += " -std=c++17"
+        #EXTRA_ENV_COMPILE_ARGS += " -std=c++17"
         EXTRA_ENV_COMPILE_ARGS += (
             " -fvisibility=hidden -fno-wrapv -fno-exceptions"
         )
