diff -up grpc-1.65.0/CMakeLists.txt.1~ grpc-1.65.0/CMakeLists.txt
--- grpc-1.65.0/CMakeLists.txt.1~	2024-06-28 05:26:05.000000000 +0200
+++ grpc-1.65.0/CMakeLists.txt	2024-06-29 21:47:00.591873364 +0200
@@ -4695,7 +4695,6 @@ foreach(_hdr
 endforeach()
 
 endif()
-if(gRPC_BUILD_TESTS)
 
 add_library(grpc++_test_config
   test/cpp/util/test_config_cc.cc
@@ -4745,7 +4744,12 @@ target_link_libraries(grpc++_test_config
   gpr
 )
 
-
+if(gRPC_INSTALL)
+  install(TARGETS grpc++_test_config EXPORT gRPCTargets
+    RUNTIME DESTINATION ${gRPC_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${gRPC_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${gRPC_INSTALL_LIBDIR}
+  )
 endif()
 if(gRPC_BUILD_TESTS)
 
@@ -15817,7 +15821,6 @@ target_link_libraries(grpc_byte_buffer_r
 
 
 endif()
-if(gRPC_BUILD_TESTS)
 
 add_executable(grpc_cli
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/reflection/v1alpha/reflection.pb.cc
@@ -15866,7 +15869,14 @@ target_link_libraries(grpc_cli
 )
 
 
+if(gRPC_INSTALL)
+  install(TARGETS grpc_cli EXPORT gRPCTargets
+    RUNTIME DESTINATION ${gRPC_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${gRPC_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${gRPC_INSTALL_LIBDIR}
+  )
 endif()
+
 if(gRPC_BUILD_TESTS)
 
 add_executable(grpc_completion_queue_test
